statsmodels/tools/_testing.py

```diff
@@ -183,3 +185,24 @@ def check_predict_types(results):
         cls = pd.Series if predicted.ndim == 1 else pd.DataFrame
         predicted_expected = cls(predicted, index=exog_index)
         assert_equal(predicted_expected, predicted_pandas)
+
+
+def try_and_xfailif(condition, reason):
```

Comment:
could you rename to `allowfailif` (to be testing-framework-like) of `allow_fail_if` to be Python-like.